### DISCUSSION TITLE
하나의 데이터를 모든 룸의 유저가 공유해 사용할 수 있는지 알고 싶습니다.

### DISCUSSION THREAD

**Discussion by LeeYongSu1** [naverz/zepeto-studio-kor#688] [5. Q&A]
World Ver: 1.3.0

App Ver: 3.12.1

구현하려는 기능:
유저가 특정 오브젝트를 클릭 하면 클릭 된 횟수를 UI에 띄우고, 이 횟수를 같은 룸의 유저 및 다른 룸에 접속 해 있는 유저도 볼 수 있거나 클릭하면 합산 되는 기능을 구현하려고 합니다.

schemas에 number Type의 schema를 정의해 데이터를 저장하면 모든 룸에 동기화 되는 건지, 하나의 데이터를 모든 룸에서는 공유하여 사용할 수 없는 건지 궁금합니다.

이 방법 외의 다른 하나의 데이트를 모든 룸의 유저가 공유하여 사용할 수 있는 방법이 있다면 어떻게 하면 되는지 알고 싶습니다.
감사합니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 말씀하신 대로 schemas에 number Type의 schema를 정의해 데이터를 저장하면 룸에 동기화 됩니다.
다만 각 클라이언트에서 처음에 서버 데이터를 받아오고, 클라이언트에서 발생한 변경 사항을 다시 서버로 업데이트 보내주고 , 
서버는 변경 사항이 발생했을때 다른 클라이언트에게 뿌려주는 코드를 작성해야 합니다.

아래 Room Message관련 가이드 및 영상 가이드를 참고하여 주세요~ 
감사합니다. 

https://studio.zepeto.me/kr/guides/multiplay-room-message
https://studio.zepeto.me/guides/multiplay-tutorial

**Reply by LeeYongSu1** [reply]
답변 감사합니다. 추가적으로 서버 데이터는 비우는 작업을 따로 하지 않으면 계속 유지가 되나요?

**Reply by Zepeto-TechEvangelist** [reply]
답변에 보충 설명 드립니다.
동일한 룸에 접속한 사람들끼리는 데이터를 주고 받을 수 있지만,
모든 룸이 데이터를 공유하기는 어렵습니다.

서버 데이터를 DataStorage를 통해서 저장하시면 월드 배포 후에는 ZEPETO DB에 저장되어서 데이터가 유지됩니다.
https://studio.zepeto.me/kr/guides/multiplay-game-datastorage
